modules/tensor_mechanics/src/actions/TensorMechanicsAction.C

```diff
@@ -277,116 +372,53 @@ TensorMechanicsAction::act()
     }
   }
 
-  // Add variables (optional)
-  else if (_current_task == "add_variable" && getParam<bool>("add_variables"))
+  // Add variables
+  else if (_current_task == "add_variable")
   {
-    auto params = _factory.getValidParams("MooseVariable");
-    // determine necessary order
-    const bool second = _problem->mesh().hasSecondOrderElements();
+    if (getParam<bool>("add_variables"))
+    {
+      auto params = _factory.getValidParams("MooseVariable");
+      // determine necessary order
+      const bool second = _problem->mesh().hasSecondOrderElements();
 
-    params.set<MooseEnum>("order") = second ? "SECOND" : "FIRST";
-    params.set<MooseEnum>("family") = "LAGRANGE";
-    if (isParamValid("scaling"))
-      params.set<std::vector<Real>>("scaling") = {getParam<Real>("scaling")};
+      params.set<MooseEnum>("order") = second ? "SECOND" : "FIRST";
+      params.set<MooseEnum>("family") = "LAGRANGE";
+      if (isParamValid("scaling"))
+        params.set<std::vector<Real>>("scaling") = {getParam<Real>("scaling")};
 
-    // Loop through the displacement variables
-    for (const auto & disp : _displacements)
+      // Loop through the displacement variables
+      for (const auto & disp : _displacements)
+      {
+        // Create displacement variables
+        _problem->addVariable("MooseVariable", disp, params);
+      }
+    }
+
+    // Homogenization scalar
+    if (_lk_homogenization)
     {
-      // Create displacement variables
-      _problem->addVariable("MooseVariable", disp, params);
+      InputParameters params = _factory.getValidParams("MooseVariable");
+      params.set<MooseEnum>("family") = "SCALAR";
+      params.set<MooseEnum>("order") = TensorMechanicsAction::_order_mapper.at(
```

Comment:
You should be able to assign the integer value of the order here directly without the `_order_mapper` machinery.

Reply:
Through the magic of https://github.com/idaholab/moose/blob/a6076d579c4ea10b0c3bdb7d7561334faf24c241/framework/include/utils/MooseEnum.h#L105

Reply:
Cool.